From 03b81e8f44c9a94291b07597af35b4f3aa16579f Mon Sep 17 00:00:00 2001
From: Xiao Tang <xtang@intel.com>
Date: Mon, 8 Jun 2026 10:56:36 +0800
Subject: [PATCH] Custom patch to support range search params (rebased onto
 #5235)

---
 faiss/IndexIDMap.cpp | 28 ++++++++++++++++++++++++----
 1 file changed, 24 insertions(+), 4 deletions(-)

diff --git a/faiss/IndexIDMap.cpp b/faiss/IndexIDMap.cpp
index 1a86c8dc0..a0d4dae60 100644
--- a/faiss/IndexIDMap.cpp
+++ b/faiss/IndexIDMap.cpp
@@ -258,11 +258,31 @@ void IndexIDMapTemplate<IndexT>::range_search(
         RangeSearchResult* result,
         const SearchParameters* params) const {
     if (params) {
-        SearchParameters internal_search_parameters;
-        IDSelectorTranslated id_selector_translated(id_map, params->sel);
-        internal_search_parameters.sel = &id_selector_translated;
+        IDSelectorTranslated this_idtrans(this->id_map, nullptr);
+        ScopedSelChange sel_change;
+        IDGrouperTranslated this_idgrptrans(this->id_map, nullptr);
+        ScopedGrpChange grp_change;
+
+        if (params->sel) {
+            auto idtrans = dynamic_cast<const IDSelectorTranslated*>(params->sel);
+
+            if (!idtrans) {
+                auto params_non_const = const_cast<SearchParameters*>(params);
+                this_idtrans.sel = params->sel;
+                sel_change.set(params_non_const, &this_idtrans);
+            }
+        }
+
+        if (params->grp) {
+            auto idtrans = dynamic_cast<const IDGrouperTranslated*>(params->grp);
 
-        index->range_search(n, x, radius, result, &internal_search_parameters);
+            if (!idtrans) {
+                auto params_non_const = const_cast<SearchParameters*>(params);
+                this_idgrptrans.grp = params->grp;
+                grp_change.set(params_non_const, &this_idgrptrans);
+            }
+        }
+        index->range_search(n, x, radius, result, params);
     } else {
         index->range_search(n, x, radius, result);
     }
-- 
2.34.1

